src/Product/SearchProvider.php

```diff
@@ -542,16 +537,15 @@ private function hideUselessFacets(array $facets, $totalProducts)
                 // There are two filters displayed
                 $usefulFiltersCount > 1
                 ||
-                /*
-                 * There is only one fitler and the
-                 * magnitude is different than the
-                 * total products
-                 */
+                // There is only one filter, but it reduces the product selection
                 (
                     count($facet->getFilters()) === 1
                     && $totalFacetProducts < $totalProducts
                     && $usefulFiltersCount > 0
                 )
+                ||
+                // There is only one filter, but it's availability or highlights
+                ($usefulFiltersCount === 1 && ($facet->getType() == 'availability' || $facet->getType() == 'highlights'))
```

Comment:
This is a bug, we must check if there is at least one useful filter, or it will display empty blocks on a page when there no products.